From f395144cff08e90e8b03742e846b95614d474f10 Mon Sep 17 00:00:00 2001
From: pierre-24 <pierreb24@gmail.com>
Date: Fri, 5 Jan 2018 17:41:44 +0100
Subject: [PATCH 1/4] Increase constants

---

diff --git a/DALTON/include/ccl1rsp.h b/DALTON/include/ccl1rsp.h
index 15cbed8..3dd8b5d 100644
--- a/DALTON/include/ccl1rsp.h
+++ b/DALTON/include/ccl1rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXZLBL
-      PARAMETER ( MAXZLBL = 100 )
+      PARAMETER ( MAXZLBL = 999 )
       LOGICAL LORXLRZ, LL1OPN
       INTEGER NLRZLBL,ISYLRZ,ISYOFZ
       CHARACTER*8 LRZLBL
diff --git a/DALTON/include/ccl2rsp.h b/DALTON/include/ccl2rsp.h
index ad1b335..827688f 100644
--- a/DALTON/include/ccl2rsp.h
+++ b/DALTON/include/ccl2rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXL2LBL
-      PARAMETER ( MAXL2LBL = 240 )
+      PARAMETER ( MAXL2LBL = 999 )
 
       LOGICAL LL2OPN
       INTEGER NL2LBL, ISYOFL2
diff --git a/DALTON/include/ccl3rsp.h b/DALTON/include/ccl3rsp.h
index 281b293..52773f5 100644
--- a/DALTON/include/ccl3rsp.h
+++ b/DALTON/include/ccl3rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXL3LBL
-      PARAMETER ( MAXL3LBL = 120 )
+      PARAMETER ( MAXL3LBL = 999 )
 
       LOGICAL LL3OPN
       INTEGER NL3LBL, ISYOFL3
diff --git a/DALTON/include/cclc1rsp.h b/DALTON/include/cclc1rsp.h
index 9e01d43..ece2d21 100644
--- a/DALTON/include/cclc1rsp.h
+++ b/DALTON/include/cclc1rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXLC1LBL
-      PARAMETER ( MAXLC1LBL = 60 )
+      PARAMETER ( MAXLC1LBL = 999 )
       INTEGER NLC1LBL,ISYLC1,ISYOFLC1,ILC1CAU
       CHARACTER*8 LBLLC1
       LOGICAL LLC1OPN
diff --git a/DALTON/include/cco1rsp.h b/DALTON/include/cco1rsp.h
index ba3d048..a8bcf2c 100644
--- a/DALTON/include/cco1rsp.h
+++ b/DALTON/include/cco1rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXO1LBL
-      PARAMETER ( MAXO1LBL = 120 )
+      PARAMETER ( MAXO1LBL = 999 )
 
       LOGICAL LO1OPN
       INTEGER NO1LBL,ISYOFO1
diff --git a/DALTON/include/cco2rsp.h b/DALTON/include/cco2rsp.h
index e1c0fd4..2906439 100644
--- a/DALTON/include/cco2rsp.h
+++ b/DALTON/include/cco2rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXO2LBL
-      PARAMETER ( MAXO2LBL = 120 )
+      PARAMETER ( MAXO2LBL = 999 )
 
       LOGICAL LO2OPN
       INTEGER NO2LBL,ISYOFO2
diff --git a/DALTON/include/cco3rsp.h b/DALTON/include/cco3rsp.h
index 30937ae..a7ee574 100644
--- a/DALTON/include/cco3rsp.h
+++ b/DALTON/include/cco3rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXO3LBL
-      PARAMETER ( MAXO3LBL = 240 )
+      PARAMETER ( MAXO3LBL = 999 )
 
       LOGICAL LO3OPN
       INTEGER NO3LBL,ISYOFO3
diff --git a/DALTON/include/ccr1rsp.h b/DALTON/include/ccr1rsp.h
index f3dbad8..bb66bba 100644
--- a/DALTON/include/ccr1rsp.h
+++ b/DALTON/include/ccr1rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXTLBL
-      PARAMETER ( MAXTLBL = 500 )
+      PARAMETER ( MAXTLBL = 999 )
       LOGICAL LORXLRT(MAXTLBL), LR1OPN
       INTEGER NLRTLBL,  ISYLRT(MAXTLBL),ISYOFT(8)
       INTEGER NLRTHFLBL,ISYLRTHF(MAXTLBL)
diff --git a/DALTON/include/ccr2rsp.h b/DALTON/include/ccr2rsp.h
index 8f645d7..35445be 100644
--- a/DALTON/include/ccr2rsp.h
+++ b/DALTON/include/ccr2rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXT2LBL
-      PARAMETER ( MAXT2LBL = 120 )
+      PARAMETER ( MAXT2LBL = 999 )
 
       LOGICAL LR2OPN
       INTEGER NR2TLBL, ISYOFT2
diff --git a/DALTON/include/ccr2trsp.h b/DALTON/include/ccr2trsp.h
index 4b308cc..c736811 100644
--- a/DALTON/include/ccr2trsp.h
+++ b/DALTON/include/ccr2trsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXT2LBL
-      PARAMETER ( MAXT2LBL = 120 )
+      PARAMETER ( MAXT2LBL = 999 )
       INTEGER NR2TLBL,ISYAR2T,ISYBR2T,ISYOFT2
       CHARACTER*8 LBLAR2T, LBLBR2T
       LOGICAL LR2OPN
diff --git a/DALTON/include/ccr3rsp.h b/DALTON/include/ccr3rsp.h
index 16f11b5..04eb4c8 100644
--- a/DALTON/include/ccr3rsp.h
+++ b/DALTON/include/ccr3rsp.h
@@ -1,5 +1,5 @@
       INTEGER MAXT3LBL
-      PARAMETER ( MAXT3LBL = 240 )
+      PARAMETER ( MAXT3LBL = 999 )
 
       LOGICAL LR3OPN
       INTEGER NR3TLBL, ISYOFT3
diff --git a/DALTON/include/ccsdio.h b/DALTON/include/ccsdio.h
index d854029..d7be56b 100644
--- a/DALTON/include/ccsdio.h
+++ b/DALTON/include/ccsdio.h
@@ -1,6 +1,6 @@
       INTEGER MAXSIM
 
-      PARAMETER(MAXSIM = 540)
+      PARAMETER(MAXSIM = 999)
 
       LOGICAL DUMPCD

-- 
1.9.1


From f395144cff08e90e8b03742e846b95614d474f10 Mon Sep 17 00:00:00 2001
From: pierre-24 <pierreb24@gmail.com>
Date: Fri, 5 Jan 2018 17:41:44 +0100
Subject: [PATCH 2/4] .GAMALL outputs in .PROP

---
 DALTON/rsp/rspcr3.F       | 46 +++++++++++++++++++++++++++++-----------------
 
diff --git a/DALTON/rsp/rspcr3.F b/DALTON/rsp/rspcr3.F
index f7391ab..62fdeec 100644
--- a/DALTON/rsp/rspcr3.F
+++ b/DALTON/rsp/rspcr3.F
@@ -733,12 +733,14 @@ C
             WRITE(LUPRI,'(A,8(A1),F16.4)') 
      *           '@ gamma(',DIPOP(I),';',DIPOP(I),',',DIPOP(J),
      *           ',',DIPOP(J),')',GAMMA(I,I,J,J)
-
-            CALL WRIPRO(GAMMA(I,I,J,J),"CR-SCF/DFT",4,
-     *                  DIQMMM(I),DIQMMM(I),
-     *                  DIQMMM(J),DIQMMM(J),
-     *                  FREQB,FREQC,FREQD,
-     *                  1,0,0,0)
+            
+            IF (.NOT. GAMALL) THEN
+               CALL WRIPRO(GAMMA(I,I,J,J),"CR-SCF/DFT",4,
+     *                     DIQMMM(I),DIQMMM(I),
+     *                     DIQMMM(J),DIQMMM(J),
+     *                     FREQB,FREQC,FREQD,
+     *                     1,0,0,0)
+            END IF
          END IF
          IF (I.NE.J) THEN
             IF (GAMMA(I,J,J,I) .EQ. D0) THEN
@@ -747,12 +749,14 @@ C
                WRITE(LUPRI,'(A,8(A1),F16.4)') 
      *              '@ gamma(',DIPOP(I),';',DIPOP(J),',',DIPOP(J),
      *           ',',DIPOP(I),')',GAMMA(I,J,J,I)
-            CALL WRIPRO(GAMMA(I,J,J,I),"CR-SCF/DFT",4,
-     *                  DIQMMM(I),DIQMMM(J),
-     *                  DIQMMM(J),DIQMMM(I),
-     *                  FREQB,FREQC,FREQD,
-     *                  1,0,0,0)
-
+            
+               IF (.NOT. GAMALL) THEN
+                  CALL WRIPRO(GAMMA(I,J,J,I),"CR-SCF/DFT",4,
+     *                        DIQMMM(I),DIQMMM(J),
+     *                        DIQMMM(J),DIQMMM(I),
+     *                        FREQB,FREQC,FREQD,
+     *                        1,0,0,0)
+               END IF
             END IF
             IF (GAMMA(I,J,I,J) .EQ. D0) THEN
                IZERO = IZERO + 1
@@ -760,11 +764,14 @@ C
                WRITE(LUPRI,'(A,8(A1),F16.4)') 
      *              '@ gamma(',DIPOP(I),';',DIPOP(J),',',DIPOP(I),
      *           ',',DIPOP(J),')',GAMMA(I,J,I,J)
-            CALL WRIPRO(GAMMA(I,J,I,J),"CR-SCF/DFT",4,
-     *                  DIQMMM(I),DIQMMM(J),
-     *                  DIQMMM(I),DIQMMM(J),
-     *                  FREQB,FREQC,FREQD,
-     *                  1,0,0,0)
+            
+               IF (.NOT. GAMALL) THEN
+                  CALL WRIPRO(GAMMA(I,J,I,J),"CR-SCF/DFT",4,
+     *                        DIQMMM(I),DIQMMM(J),
+     *                        DIQMMM(I),DIQMMM(J),
+     *                        FREQB,FREQC,FREQD,
+     *                        1,0,0,0)
+               END IF
             END IF
          END IF
          GAMAVE=GAMAVE + 
@@ -790,6 +797,11 @@ C
                   WRITE(LUPRI,'(A,8(A1),F18.8)') 
      *                 '@ gamma(',DIPOP(I),';',DIPOP(J),',',DIPOP(K),
      *                 ',',DIPOP(L),')',GAMMA(I,J,K,L)
+                  CALL WRIPRO(GAMMA(I,J,K,L),"CR-SCF/DFT",4,
+     *                        DIQMMM(I),DIQMMM(J),
+     *                        DIQMMM(K),DIQMMM(L),
+     *                        FREQB,FREQC,FREQD,
+     *                        1,0,0,0)
                 END IF
               END DO
             END DO
-- 
1.9.1



From 0b9901597d28ad3fed457dae907ee03563dc2430 Mon Sep 17 00:00:00 2001
From: pierre-24 <pierreb24@gmail.com>
Date: Fri, 5 Jan 2018 18:38:04 +0100
Subject: [PATCH 3/4] add beta to .PROP

---

diff --git a/DALTON/rsp/rspvec.F b/DALTON/rsp/rspvec.F
index ed77639..316dd81 100644
--- a/DALTON/rsp/rspvec.F
+++ b/DALTON/rsp/rspvec.F
@@ -953,6 +953,7 @@ C CALCULATION OF FIRST HYPERPOLARIZABILITIES AND RELATED PROPERTIES
 C
       CHARACTER*1 ALABP, BLABP, CLABP
       CHARACTER*8 BLANK, LABEL
+      CHARACTER*7 BLANK_CHARS
       DIMENSION CMO(*),UDV(*),PV(*),FOCK(*),FC(*),FV(*),FCAC(*),H2AC(*)
       DIMENSION XINDX(*),VECB(*),VECC(*),WRK(*)
       DIMENSION SPNFC1(*), SPNFC2(*), SPNFC(*), SPNSD1(*), SPNSD2(*),
@@ -1687,6 +1688,14 @@ C
      &              '@ B-freq =',BQRFR(IBFR),'  C-freq =',CQRFR(ICFR),
      &                   '     beta(',ALABP,';',BLABP,',',CLABP,')',
      &                        ' =',-HYPVAL(IAOP)
+                           CALL WRIPRO(-HYPVAL(IAOP),"QR-SCF/DFT",3,
+     *                                 ALABP // '       ',
+     *                                 BLABP // '       ',
+     *                                 CLABP // '       ',
+     *                                 ALABP // '       ',
+     *                                 BQRFR(IBFR),CQRFR(ICFR),
+     *                                 CQRFR(ICFR),
+     *                                 1,0,0,0)
 C      hjaaj Oct 2001: note that
 C            beta = <<mu; -mu, -mu>> = -<<r; r, r>>
                               END IF
-- 
1.9.1



From 0b9901597d28ad3fed457dae907ee03563dc2430 Mon Sep 17 00:00:00 2001
From: pierre-24 <pierreb24@gmail.com>
Date: Fri, 5 Jan 2018 18:38:04 +0100
Subject: [PATCH 4/4] Outputs more information about missing memory

---

diff --git a/DALTON/cc/cc_2hyp.F b/DALTON/cc/cc_2hyp.F
index e761c69..85c1393 100644
--- a/DALTON/cc/cc_2hyp.F
+++ b/DALTON/cc/cc_2hyp.F
@@ -257,6 +257,9 @@ c /* deck CC_2HYP */
       LEND0    = LWORK - KEND0
 
       IF (LEND0 .LT. 0) THEN
+       WRITE (LUPRI,*) 'Insufficient memory in CC_2HYP. (1)'
+       WRITE (LUPRI,*) 'available:',LWORK
+       WRITE (LUPRI,*) '   needed:',KEND0
        CALL QUIT('Insufficient memory in CC_2HYP. (1)')
       END IF
 
@@ -1243,6 +1246,9 @@ c    &                  LBLOPR(IOP(D)),.FALSE.,SIGN*DCRFR(IFREQ),ISYM2)
         END IF
 
         IF (LEND1 .LT. 0) THEN
+          WRITE (LUPRI,*) 'Insufficient memory in CC_2HYP. (3)'
+          WRITE (LUPRI,*) 'available:',LWORK
+          WRITE (LUPRI,*) '   needed:',KEND0
           CALL QUIT('Insufficient memory in CC_2HYP. (3)')
         END IF

-- 
1.9.1

